packages/dds/tree/docs/wip/schema-evolution.md

```diff
@@ -0,0 +1,583 @@
+     For example, imagine a digital whiteboard app that renders various types of content to the screen.
+     The developer chooses "Ink", "Sticky Note" and "Unknown" (the catch-all) as the possible node types allowed in the whiteboard canvas field.
+     To render an "Unknown", the app makes a gray box that says "This content is not supported, please update to the latest version of the application", but the app otherwise is fully functional as it has been (it is _not_ an error).
+     In a future version of the application, the possible node types are increased to "Ink", "Sticky Note", "Unknown" and "Image".
+     A version of the app with the original view schema will interpret Images as "Unknown" and render the gray box.
+     Therefore the new version, with the new view schema, can go ahead and upgrade the stored schema as well without worrying about doing the staged rollout process.
+
+     Clearly, when this strategy makes sense for the application, it immensely reduces the amount of work required to add a new node type.
+     However, it must be employed preemptively.
+     The developer needs to include the "Unknown" type ahead of time, anticipating that "there might be some new stuff here someday".
+   * To mitigate that problem, another option would be to have the "Unknown" type be automatic - or "opt-out" rather than "opt-in".
+     That is, all applications must handle the "Unknown" type in every field, unless they intentionally make the decision not to.
+   * A third approach would be to automatically filter out unrecognized node types where applicable.
+     In the example above, the original client would simply not be able to observe that there are Image nodes in the canvas field - the field would appear shorter than it does to the upgraded clients if there are images in it.
+     This might be appropriate for sequence fields, but would not work for required fields because a required field must not appear empty.
+
+     > TODO: This is one of Noah's off-the-cuff ideas. Whether or not filtering makes sense for optional fields (or at all) is debatable.
+
+     This strategy could be used in combination with the "Unknown" type - if Unknown is in the field's set of allowed types, then the foreign type will present as an Unknown, otherwise it will be filtered.
+
+1. **Remove an allowed type.**
+   This scenario can employ the same strategy as "Add an allowed type" above, namely, the "Unknown" catch-all type.
+   The newer schema can simply add the "Unknown" type to the set of allowed types if it doesn't already exist.
+   Then the removed type, when present in the document, will present as an "Unknown".
+
+   This does not require the developer to plan ahead - "Unknown" can be added to the new view schema when the field is first removed, rather than having to be added preemptively in the original schema as is the case when adding an allowed type.
```

Comment:
Does this describe existing tool, or just enumerates "we can build it if you ask"?